src/wallet/mod.rs

```diff
@@ -4095,4 +4130,39 @@ pub(crate) mod test {
             "when there's no internal descriptor it should just use external"
         );
     }
+
+    #[test]
+    fn test_taproot_psbt_populate_tap_key_origins() {
```

Comment:
In this test we are testing the script path key origin insertion.. Probably that's because you didn't had the script path tr descriptor at that time.. But now we can test script path key origin insertion using `get_test_tr_with_taptree_xprv` it seems?

Reply:
It's kinda testing two different things, but I agree it makes sense to test that part as well. Here we are testing the key origins for key spend (because the leaf hash vec is empty), I'll add one more test for when the key appears in multiple leaves to make sure we are accounting for all of them